parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriterV2.java

```diff
@@ -64,15 +65,17 @@ public ColumnWriterV2(
       ColumnDescriptor path,
       PageWriter pageWriter,
       ParquetProperties parquetProps,
-      int pageSize) {
+      int pageSize,
+      ByteBufferAllocator allocator) {
     this.path = path;
     this.pageWriter = pageWriter;
     resetStatistics();
 
-    this.repetitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxRepetitionLevel()), MIN_SLAB_SIZE, pageSize);
-    this.definitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxDefinitionLevel()), MIN_SLAB_SIZE, pageSize);
+    this.repetitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxRepetitionLevel()), MIN_SLAB_SIZE, pageSize, allocator);
+    this.definitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxDefinitionLevel()), MIN_SLAB_SIZE, pageSize, allocator);
 
     int initialSlabSize = CapacityByteArrayOutputStream.initialSlabSizeHeuristic(MIN_SLAB_SIZE, pageSize, 10);
+    // TODO - Should I be passing an allocator here rather than just letting it use the one stored in ParquetProperties
```

Comment:
Really this means that every class (including this one) that has a ParquetProperties in its constructor parameters should not take an allocator as well.
Do you want to remove the Allocator parameter when there's a ParquetProperties and this comment?